### PR TITLE
added the language parameter for the retrieve api

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**

### CODE REVIEW

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java [hunk 1]
```diff
@@ -136,5 +147,35 @@ public String computeEtag(long timestamp, String format, String version){
         return DigestUtils.md5Hex(String.format("%s:%s:%s", timestamp, format, version));
     }
 
+    @SuppressWarnings("unchecked")
+	private void processLanguage(Entity entity, String languages) {
+		if(languages==null) return;
+		List<String> languagesList = Arrays.asList(languages.split(",", -1));
+    	List<Field> entityFields = EntityUtils.getAllFields(entity.getClass());
+		for(Field field : entityFields) {
+			if(field.getType().isAssignableFrom(Map.class)) {
```
Comment (gsergiu) [inline entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java]: notation is a map but not multilingual. 
Use EntityFieldsTypes.isMultilingual() instead, but also check if the field value is not null before checking the type

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java [hunk 2]
```diff
@@ -136,5 +147,35 @@ public String computeEtag(long timestamp, String format, String version){
...
+				Map<String, Object> currentFieldValue;
+				try {
+					currentFieldValue = (Map<String,Object>) entity.getFieldValue(field);
+				} catch (IllegalArgumentException | IllegalAccessException e) {
+					throw new FunctionalRuntimeException("An exception occured during accessing the entity field: "+field.getName(), e);
+				}
+				if(currentFieldValue==null) continue;
+				Map<String,Object> newFieldValue = new HashMap<>();
+				for(Map.Entry<String,Object> mapEntry : currentFieldValue.entrySet()) {
+					if(languagesList.contains(mapEntry.getKey())) {
```
Comment (gsergiu) [inline entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java]: do not filter out URLs, which are available in the empty keys ""

**Review by gsergiu — CHANGES_REQUESTED** [review]
please address the review comments

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java]: Improved.

File: entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java (hunk 2 above)
Comment (SrdjanStevanetic) [reply entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java]: Improved.

**Pull request merged**